src/App.js

```diff
@@ -14,11 +14,61 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    selected: 'Jam',
+  }
 
-export default App;
+   selectItem = (event) => {
+     const itemContainer = event.target.closest('li');
+
+     this.setState({
+       selected: itemContainer.children[0].innerText,
+     });
+   }
+
+   clearSelected = () => {
+     this.setState({ selected: '' });
+   }
+
+   render() {
+     const { selected } = this.state;
+
+     return (
+       <div className="App">
+         <div className="title">
+           <h1>
+             {selected
+               ? `${selected} is selected `
+               : 'No goods selected'}
+           </h1>
+           {selected && (
+           <button type="button" onClick={() => this.clearSelected()}>
+             X
+           </button>
+           )}
+         </div>
+         <ul className="list">
+           {goodsFromServer.map(item => (
+             <li
+               key={item}
+               className={selected === item ? 'selected' : ''}
+             >
+               <span>
+                 {item}
+               </span>
+               {selected !== item && (
+               <button
+                 type="button"
+                 onClick={event => (this.selectItem(event))}
```

Comment:
```suggestion
                 onClick={event => (this.selectItem(item))}
```
Передавай туда item, зачем передавать event, чтобы потом в нем искать innerText?

Reply:
yep, fixed